src/app/app.module.ts

```diff
@@ -1,16 +1,45 @@
 import { NgModule } from '@angular/core';
 import { BrowserModule } from '@angular/platform-browser';
-
 import { AppRoutingModule } from './app-routing.module';
+import { HttpClientModule } from "@angular/common/http";
+import { AngularSvgIconModule } from 'angular-svg-icon';
+import { CommonModule } from '@angular/common';
+import { FormsModule, ReactiveFormsModule } from '@angular/forms';
+import { MatDialogModule } from '@angular/material/dialog';
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { MatInputModule } from '@angular/material/input';
+import { MatButtonModule } from '@angular/material/button';
+import { NoopAnimationsModule } from "@angular/platform-browser/animations";
+
 import { AppComponent } from './app.component';
+import { HeaderComponent } from './components/header/header.component';
+import { ArticleComponent } from './components/article/article.component';
+import { SideBarComponent } from './components/side-bar/side-bar.component';
+import { HomePageComponent } from "./pages/home-page/home-page.component";
+import { ArticleDialogComponent } from './components/article-dialog/article-dialog.component';
```

Comment:
lets split into:
Shared module with shared components, like material
Feature module, in this case blog/posts